### ISSUE TITLE
S3 - ver82fly2 - 761e575c-95dc-4354-a81e-ae41ac27a21c

### ISSUE THREAD

**Issue by m3d** [robotika/subt-artf#34]
Jedna drona A250R s testem padu do diry. Tam spadne OK, ale problem je Z-ova souradnice po tom padu (!)

![s3-A250R-height-profile](https://user-images.githubusercontent.com/5664353/95672366-7923e300-0ba0-11eb-8483-e468ea87372b.png)

mota se tam dokolecka a "fiktivne stoupa", ale ve skutecnosti poletuje stale na stejnem miste. To by komplikovalo i pripadny 3D navrat. Mame logovany i ten barometr?

**Comment by jisa** [comment]
Barometr by mohl byt logovany na strane ROSu. Do tech logu jsem se nikdy nedival. Sami si ho nepreposilame. Mohli bychom.

Pri poklesu dirou sebou dole drona triskla o zed. Je mozne, ze se tam Kalman rozbil a uz se nevzpamatoval.

**Comment by zwn** [comment]
Už dříve jsme pozorovali, že kontakt drony s kameny rozbíjí lokalizaci natolik, že se může i ztratit. Pokud tam byl tvrdší kontakt se zemí, tak by to považoval za velkého kandidáta na zdroj problémů.

**Comment by jisa** [comment]
Doufal jsem, ze to rozbiti lokalizace bude skokove. Pak by se ta drona mela dokazat vratit aspon k tomu mistu narazu. To se nestalo. To by znamenalo, ze se v Kalmanovi rozbiji nejen xyz, ktere uz se vzpamatovat nemuzou, ale i rychlosti a/nebo zrychleni, ktere by se vzpamatovat mely.

Krome barometru, ktery reportuje pro cloveka necitelny tlak, bych logoval i vystup prevodu do "pose".

Vrtam do Kalmana, jestli ho dokazu presvedcit, ze ma te vyskove souradnici opravdu opravdu verit. Porad nechapu, proc to tak neni. Jedna z moznosti je, ze potize zpusobuji nulove kovariacni matice reportovane z imu.

**Comment by jisa** [comment]
Jo, tak kovariacni matice by napr. pro linearni zrychleni mela obsahovat 1e-4:
https://github.com/osrf/subt/blob/840975255690efaa16ca2d851ff8e606fc126648/submitted_models/ssci_x4_sensor_config_1/model.sdf#L184

Protoze tam ale je nula, UKF ji nahradi za 1e-9:
https://github.com/cra-ros-pkg/robot_localization/blob/f6bf556407945e7b192c78959c51d2f10f62e488/src/ukf.cpp#L199

To je o pet radu moc nizke cislo. Dovedu si predstavit, ze to prehlasuje ostatni odhady / senzory.

Jdu vymyslet, co s tim a zkoumat, jestli to neco udela.

**Comment by m3d** [comment]
Pridal jsem poznamku s grafem z barometru k PR:
https://github.com/robotika/osgar/pull/676#issuecomment-706705644
skoro mam cukani (stejne jako to delame s IMU) vzit Z pouze z barometru a zbytek ignorovat. Minimalne uz zase zacina mit smysl sledovani 3D trace.
Jeste u techto propadu asi neni ani validni pozice reportovanych artefaktu ... necham na @jisa, ze by se idealne zlepsila ta `pose3d`.
dik m.

**Comment by jisa** [comment]
Nemel bys k takovym grafum i popisky, co znamena ktera cara a co ktera osa?

Kdyz se mi nepodari presvedcit kalmana, ze ma Z verit, muzeme to "ohackovat" dalsim modulem, ktery vystup prepise. Zatim se mi do toho moc nechce, protoze lepsi Kalman zlepsi i dalsi osy a tomu bych dal prednost.

**Comment by m3d** [comment]
jj, labely jsou vzdy muj problem :( ... je tam Z, relativne laser dolu a nahoru a nove altitude

**Comment by jisa** [comment]
Asi jsem nasel chybu v nasem kodu, kvuli ktere Kalman altimetru nedoveruje, nebo ho dokonce nepouziva vubec. Chvili do toho budu dal vrtat a pak se, snad jeste behem vecera, ozvu.
